src/initializer.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2021 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Olm from "@matrix-org/olm";
+import olmWasmPath from "@matrix-org/olm/olm.wasm?url";
+import { Integrations } from "@sentry/tracing";
+import i18n from "i18next";
+import { initReactI18next } from "react-i18next";
+import LanguageDetector from "i18next-browser-languagedetector";
+import Backend from "i18next-http-backend";
+import * as Sentry from "@sentry/react";
+
+import { getUrlParams } from "./UrlParams";
+import { Config } from "./config/Config";
+import { DEFAULT_CONFIG } from "./config/ConfigOptions";
+
+enum LoadState {
+  None,
+  Loading,
+  Loaded,
+}
+
+class DependencyLoadStates {
+  olm: LoadState = LoadState.None;
+  config: LoadState = LoadState.None;
+  sentry: LoadState = LoadState.None;
+  i18n: LoadState = LoadState.None;
+
+  allDepsAreLoaded() {
+    const notLoadedDeps = Object.values(this).filter(
+      (s) => s !== LoadState.Loaded
+    );
+    return notLoadedDeps.length === 0;
```

Comment:
```suggestion
    return !Object.values(this).some(
      (s) => s !== LoadState.Loaded
    );
```

Will `allDepsAreLoaded()` also be one of the `values`?

Reply:
I was also worried but that seems to be not the case with class instances.
`obj = {a:()=>{return 0}}` would definitly be included in the values.

Reply:
Looks like its fine
![image](https://user-images.githubusercontent.com/16718859/198883334-1617ea17-5ca2-4b63-b167-4e1e2473831e.png)

Reply:
Great!